File: BookRentalSpringBoot/src/main/java/com/example/demo/models/BookModel.java
package com.example.demo.models;


public class BookModel {

    private long id;
    private String title;
    private long publishYear;
    private String author;
    private String publisher;

    public String getPublisher() {
        return publisher;
    }

    public void setPublisher(String publisher) {
        this.publisher = publisher;
    }
//    public BookModel(String title, long publishYear, String author, int numOfPages) {
//        this.title = title;
//        this.publishYear = publishYear;
//        this.author = author;
//    }

    public void setTitle(String title) {
        this.title = title;
    }

    public void setPublishYear(long publishYear) {
        this.publishYear = publishYear;
    }

    public void setAuthor(String author) {
        this.author = author;
    }

    public String getTitle() {
        return title;
    }

    public long getPublishYear() {
        return publishYear;
    }

    public String getAuthor() {
        return author;
    }


    @Override
    public String toString() {
        return "BookModel{" +
                "title='" + title + '\'' +
                ", author='" + author + '\'' +
                ", publishYear=" + publishYear +
                ", publisher='" + publisher + '\'' +
                '}';
    }
}

File: BookRentalAngular/src/app/models/Book.model.ts

export class Book {


  private _title;
  private _publishYear;
  private _author;
  private _publisher;


  constructor(title, publishYear, author, publisher) {
    this._title = title;
    this._publishYear = publishYear;
    this._author = author;
    this._publisher = publisher;
  }


  get title() {
    return this._title;
  }

  set title(value) {
    this._title = value;
  }

  get publishYear() {
    return this._publishYear;
  }

  set publishYear(value) {
    this._publishYear = value;
  }

  get author() {
    return this._author;
  }

  set author(value) {
    this._author = value;
  }

  get publisher() {
    return this._publisher;
  }

  set publisher(value) {
    this._publisher = value;
  }
}

